lib/aggregate.js

```diff
@@ -569,6 +572,79 @@ Aggregate.prototype.read = function(pref, tags) {
   return this;
 };
 
+/**
+ * Sets the readConcern level for the aggregation query.
+ *
+ * ####Example:
+ *
+ *     Model.aggregate(..).readConcern('majority').exec(callback)
+ *
+ * @param {String} level one of the listed read concern level or their aliases
+ * @see mongodb https://docs.mongodb.com/manual/reference/read-concern/
+ * @return {Aggregate} this
+ * @api public
+ */
+
+Aggregate.prototype.readConcern = function(level) {
+  if (!this.options) {
+    this.options = {};
+  }
+  readConcern.call(this, level);
+  return this;
+};
+
+/**
+ * Appends a new $redact operator to this aggregate pipeline.
+ *
+ * If 3 arguments are supplied, Mongoose will wrap them with if-then-else of $cond operator respectively
+ * If `thenExpr` or `elseExpr` is string, make sure it starts with $$, like `$$DESCEND`, `$$PRUNE` or `$$KEEP`.
+ *
+ * ####Example:
+ *
+ *     Model.aggregate(...)
+ *      .redact({
+ *        $cond: {
+ *          if: { $eq: [ '$level', 5 ] },
+ *          then: '$$PRUNE',
+ *          else: '$$DESCEND'
+ *        }
+ *      })
+ *      .exec();
+ *
+ *     // $redact often comes with $cond operator, you can also use the following syntax provided by mongoose
+ *     Model.aggregate(...)
+ *      .redact({ $eq: [ '$level', 5 ] }, '$$PRUNE', '$$DESCEND')
+ *      .exec();
+ *
+ * @param {Object} expression redact options or conditional expression
+ * @param {String|Object} [thenExpr] true case for the condition
+ * @param {String|Object} [elseExpr] false case for the condition
+ * @return {Aggregate} this
+ * @see $redact https://docs.mongodb.com/manual/reference/operator/aggregation/redact/
+ * @api public
+ */
+
+Aggregate.prototype.redact = function(expression, thenExpr, elseExpr) {
+  if (arguments.length === 3) {
+    if ((typeof thenExpr === 'string' && !thenExpr.startsWith('$$')) ||
```

Comment:
We can't use `startsWith()` right now because it isn't in Node.js 6 or earlier. Use `thenExpr.indexOf('$$') !== -1` instead.

Reply:
@vkarpov15 I tested it on node.js 4.0.0 and it works. node 0.12 does not work though.

```sh
$ nvm use 4.0.0
Now using node v4.0.0 (npm v2.14.2)
$ node --version
v4.0.0
$ node
> 'abc'.startsWith('ab')
true
> 'abc'.startsWith('aG')
false

$ nvm use 0.12.18
Now using node v0.12.18 (npm v2.15.11)
$ node --version
v0.12.18
$ node
> 'abc'.startsWith('ab')
TypeError: undefined is not a function
    at repl:1:7
```